protocols/raft/src/main/java/io/atomix/protocols/raft/zeebe/ZeebeLogAppender.java

```diff
@@ -28,8 +32,53 @@
    * @param lowestPosition lowest record position in the data buffer
    * @param highestPosition highest record position in the data buffer
    * @param data data to store in the entry
-   * @return a future which is completed with the indexed entry is committed
    */
-  CompletableFuture<Indexed<ZeebeEntry>> appendEntry(
-      long lowestPosition, long highestPosition, ByteBuffer data);
+  void appendEntry(long lowestPosition, long highestPosition, ByteBuffer data, Listener listener);
+
+  /**
+   * An append listener can observe and be notified of different events related to the append
+   * operation.
+   */
+  interface Listener {
+
+    /**
+     * Called when the entry has been written to the log.
+     *
+     * @param indexed the entry that was written to the log
+     */
+    void onWrite(Indexed<ZeebeEntry> indexed);
+
+    /**
+     * Called when the entry has been committed.
+     *
+     * @param indexed the entry that was committed
+     */
+    void onCommit(Indexed<ZeebeEntry> indexed);
+
+    /**
+     * Called if an error occurs during appending.
+     *
+     * @param error the error that occurred
+     */
+    void onError(Throwable error);
+
+    final class NoopListener implements Listener {
```

Comment:
Actually we dont use it right? So we can remove it

Reply:
I think I was using it then not and it got left over :see_no_evil: